src/lineage/entrypoint.py

```diff
@@ -180,14 +180,25 @@ def merge(repo: Repository.Repository, github_actor: str) -> Tuple[bool, List[st
     return True, conflict_file_list
 
 
-def push(repo: Repository.Repository, branch_name: str, username: str, password: str):
+def push(
+    repo: Repository.Repository, branch_name: str, username: str, password: str
+) -> bool:
     """Push changes to remote."""
-    parts: ParseResult = urlparse(repo.clone_url)
-    cred_url = parts._replace(netloc=f"{username}:{password}@{parts.netloc}").geturl()
-    logging.info("Assigning credentials for push.")
-    run([GIT, "remote", "set-url", "origin", cred_url], cwd=repo.full_name)
-    logging.info(f"Pushing {branch_name} to remote.")
-    run([GIT, "push", "--set-upstream", "origin", branch_name], cwd=repo.full_name)
+    if not repo.permissions.push:
+        logging.warning(
+            f"⚠️ ERROR! Missing 'push' permission on '{repo.owner.login}/{repo.name}'"
+        )
+        return False
+    else:
+        parts: ParseResult = urlparse(repo.clone_url)
+        cred_url = parts._replace(
+            netloc=f"{username}:{password}@{parts.netloc}"
+        ).geturl()
+        logging.info("Assigning credentials for push.")
+        run([GIT, "remote", "set-url", "origin", cred_url], cwd=repo.full_name)
```

Comment:
At some point we should replace these git commands with [GitPython](https://pypi.org/project/GitPython/).

Reply:
See #12, which I just created.

Reply:
@felddy Do you remember why you didn't leverage that when you first wrote this?

Reply:
Probably didn't know about it yet.  Not a technical reason.